tutorials/python/python-type-driven-development/tutorial.md

```diff
@@ -0,0 +1,211 @@
+
+In this article, we’ll be walking through the Python - how to add types to our code and Docstrings and perform static type checking using [mypy][mypy]. There are plenty options for static type-checking in Python, but in this tutorial we will follow [Guido van Rossum suggestion][guido-mypy].
+
+### Mypy
+
+Mypy is a third-party Python library that provides optional static type checking. Unlike other non-dynamic programming languages like Java, where the static type-checking takes place at compilation time, Mypy CLI does the type-check to a file (or a set of files), so it's up to you to decide where in your workflow the type-checking will be executed. It can be on CI or a pre-commit hook, just to mention some.
+
+To start using mypy, install it using any version of `pip` globally or in your virtual environment.
+
+```bash
+pip install mypy
+```
+
+To perform static type checking on a source file:
+
+```bash
+mypy my_file.py
+```
+
+## Primitive types
+
+Python is a multi-paradigm dynamic language, so the type of a variable is determined based on the value. This might be confusing, but it doesn't mean that Python has _weak typing_. Python is _strongly typed_. Surprised? Check this out:
+
+```python
+movie = "Die Hard"
+movie = movie + 2
+# TypeError: cannot concatenate 'str' and 'int' objects
+```
+
+That said, let’s begin this tutorial with a basic example:
+
+```python
+meat = "Ground beef"
+print(type(meat))
+# <class 'str'>
+
+weight_pounds = 0.5
+print(type(weight_pounds))
+# <class 'float'>
+```
+
+In the above example, we are using the `type` function to inspect the type representation of any variable. What if we include types in our first example code? Let’s check the following example:
+
+```python
+meat: str = "Ground beef"
+weight_pounds: str = 0.5
+```
+
+If we execute this file we expect the same output as the previous one, but we actually get the following:
+
+```
+error: Incompatible types in assignment (expression has type "float", variable has type "str")
+```
+
+Did you catch the reason? The variable `weight_pounds` was defined as `str` but we were assigning a float number to it. We can start spotting bugs in every type check.
+
+I’m pretty sure you can understand this function:
+
+```python
+def make_hamburger(meat, number_of_meats):
+   return ["bread"] + [meat] * number_of_meats + ["bread"]
+
+
+print(make_hamburger("ground beef", 2))
+# ['bread', 'ground beef', 'ground beef', 'bread']
+```
+
+Let’s see how it goes with types:
+
+```python
+from typing import List
+
+
+def make_hamburger(meat: str, number_of_meats: int) -> List[str]:
+   return ["bread"] + [meat] * number_of_meats + ["bread"]
+```
+
+What happened here? We defined the function’s arguments types: string for `meat`, integer for `number_of_meats` and the return type which is a list of string values. Did you see the import at the beginning? Complex types must be imported from the `typing` library. We will cover some of them later.
```

Comment:
Was this covered later? I think there's an example using this import, but maybe that can be explicit. I think the typing library is also mentioned at the very end of the tutorial, so that might be what confused me.

Reply:
I think it's better to remove this "We will cover some of them later." and elaborate more abot the complex types and the `typing` lib.